projects/vdk-plugins/vdk-meta-jobs/README.md

```diff
@@ -0,0 +1,159 @@
+You install a new python library (this plugin itself) and you are ready to go.
+
+Using this plugin you can specify dependencies between data jobs as a direct acyclic graph (DAG). See usage for more information.
+
+## Usage
+
+```
+pip install vdk-meta-jobs
+```
+
+Then one would create a single step and define the jobs we want to orchestrate
+
+```python
+def run(job_input):
+    jobs = [
+        {
+        "job_name": "name-of-job",
+        "team_name": "team-of-job",
+        "fail_meta_job_on_error": True or False,
+        "depends_on": [name-of-job1, name-of-job2]
+        },
+        ...
+    ]
+    MetaJobInput().run_meta_job(jobs)
+```
+
+When defining a job to be run following attributes are supported:
+* **job_name**: required, the name of the data job
+* **team_name:**: optional, the team of the data job. If omitted , it will use the meta job's team
+* **fail_meta_job_on_error**: optional, default is true. if true, the meta job will abort and fail if the orchestrated job fails, if false, meta job won't fail and continue.
+* **depends_on**: required (can be empty), list of other jobs that the orchestrated job depends on. The job will not be started until depends_on job have finished.
+
+
+### Example
+
+The following example dependency graph can be implemented with below code.
+
+
+![img_2.png](img_2.png)
+
+In this example what happens is
+* Job 1 will execute.
+* After Job 1 is completed, jobs 2,3,4 will start executing in parallel.
+* Jobs 5 and 6 will start executing after job 3 completes, but will not wait for the completion of jobs 2 and 4.
+
+
+```python
+
+from vdk.api.job_input import IJobInput
+from vdk.plugin.meta_jobs.meta_job_runner import MetaJobInput
+
+JOBS_RUN_ORDER = [
+    {
+        "job_name": "job1",
+        "team_name": "team-awesome",
+        "fail_meta_job_on_error": True,
+        "depends_on": []
+    },
+
+    {
+        "job_name": "job2",
+        "team_name": "team-awesome",
+        "fail_meta_job_on_error": True,
+        "depends_on": ["job1"]
+    },
+    {
+        "job_name": "job3",
+        "team_name": "team-awesome",
+        "fail_meta_job_on_error": True,
+        "depends_on": ["job1"]
+    },
+    {
+        "job_name": "job4",
+        "team_name": "team-awesome",
+        "fail_meta_job_on_error": True,
+        "depends_on": ["job1"]
+    },
+
+    {
+        "job_name": "job5",
+        "team_name": "team-awesome",
+        "fail_meta_job_on_error": True,
+        "depends_on": ["job3"]
+    },
+    {
+        "job_name": "job6",
+        "team_name": "team-awesome",
+        "fail_meta_job_on_error": True,
+        "depends_on": ["job3"]
+    },
+]
+
+
+def run(job_input: IJobInput) - > None:
+    MetaJobInput().run_meta_job(JOBS_RUN_ORDER)
+```
+
+### FAQ
+
+
+**Q: Will the metajob retry on Platform Error?**<br>
+A: Yes, up to N (configurable by the Control Service) attempts for each job it is orchestrating.
```

Comment:
is it going to be retried immediately, or is there any backoff

Reply:
That's not upto this plugin. This is managed by the Control Service (we are talking here about the normal re-try of a data job, nothing fancy).